### PR TITLE
Support iOS 16 Live Text

### PR DESCRIPTION
**Pull request by ba01ei**
This change allows Agrume to optionally enable iOS 16 Live Text.

Demo:

https://user-images.githubusercontent.com/1700014/192006025-a4068133-014d-479f-a911-c4db7951259e.mov

### CODE REVIEW

File: Agrume/AgrumeCell.swift [hunk 1]
```diff
@@ -60,12 +61,24 @@ final class AgrumeCell: UICollectionViewCell {
...
         imageView.image = image
+        if enableLiveText {
+          if #available(iOS 16, *) {
+            if let image = image {
+              Task {
+                await analyzeImage(image)
+              }
+            }
+          }
+        }
```
Comment (JanGorman) [inline Agrume/AgrumeCell.swift:81]: I think we can simplify this expression and get rid of some of the nesting:
```suggestion
        if #available(iOS 16, *), enableLiveText, let image = image {
          Task {
            await analyzeImage(image)
          }
        }
```

File: Agrume/AgrumeCell.swift [hunk 2]
```diff
@@ -482,4 +495,27 @@ extension AgrumeCell: UIScrollViewDelegate {
...
+    let configuration = ImageAnalyzer.Configuration([.text, .machineReadableCode])
+    do {
+      let analysis = try await analyzer.analyze(image, configuration: configuration)
+      await MainActor.run {
+        interaction.analysis = analysis
+        interaction.preferredInteractionTypes = .automatic
+      }
+    } catch {
+    }
+  }
```
Comment (JanGorman) [inline Agrume/AgrumeCell.swift:520]: What if we changed the method to not be async and instead encapsulate that inside of the method itself. Then we don't need the `await MainActor.run` dance to construct the interaction. We could then also make use of `Task { @MainActor in`

Another alternative would be to keep the method `async` but annotate it to run on the `MainActor`:

```swift
@MainActor
private func analyzeImage(_ image: UIImage) async {
  …
}
```

```suggestion
  private func analyzeImage(_ image: UIImage) {
    guard ImageAnalyzer.isSupported else {
      return
    }
    let interaction = ImageAnalysisInteraction()
    imageView.addInteraction(interaction)

    let analyzer = ImageAnalyzer()
    let configuration = ImageAnalyzer.Configuration([.text, .machineReadableCode])

    Task { @MainActor in
      do {
        let analysis = try await analyzer.analyze(image, configuration: configuration)
        interaction.analysis = analysis
        interaction.preferredInteractionTypes = .automatic
      } catch {
        print(error.localizedDescription)
      }
    }
  }
```

**Review by JanGorman — COMMENTED** [review]
Thanks for opening the PR, this looks great and it's a nice addition to the library! I added some suggestions in the PR

**Comment by ba01ei** [comment]
Thanks a lot for the feedback @JanGorman  I have updated the code based on your suggestions.

**Comment by CarlosR1993** [comment]
Hi Jan,
  

  
What is it exactly you do do you delete or edit my boyfriend text messages he sends to other women so that I don’t find out ?   because I feel that is. It right or please explain too me   
  
  
  
  
  
  
  

  
  
>

**Comment by JanGorman** [comment]
Awesome, thanks @ba01ei!

**Comment by ba01ei** [comment]
> Awesome, thanks @ba01ei!

Thanks a lot for merging @JanGorman. Do you plan to add a new release or tag some time soon?

**Comment by hyouuu** [comment]
Screen Shot 2022-10-05 at 10 14 20 PM

@JanGorman @ba01ei did you test it? Seems the dependency was not added

**Comment by JanGorman** [comment]
Interesting, it was working fine when I tried it

**Pull request merged**